### PR TITLE
Allow loading external backends

### PR DESCRIPTION
**Pull request by farizrahman4u**
This allows third parties to run keras on their backends without having to add it to keras core / mokey patch / maintain a separate fork.

### CODE REVIEW

File: keras/backend/__init__.py [hunk 1]
```diff
@@ -83,7 +81,11 @@
     sys.stderr.write('Using TensorFlow backend.\n')
     from .tensorflow_backend import *
 else:
-    raise ValueError('Unknown backend: ' + str(_BACKEND))
+    try:
+        exec('from ' + _BACKEND + ' import *')
```
Comment (fchollet) [inline keras/backend/__init__.py]: Using `exec` on arbitrary code would be extremely unsafe.

Here is one safer solution:

```python
backend_module = importlib.import_module(_BACKEND)
globals().update(backend_module.__dict__)
```

However it has one issue: it potentially pollutes the namespace of `keras.backend` with irrelevant entries (such as imports made by the module) and could override some important names such as `epsilon`, etc.

But overall the behavior is the same as with the other backends, so it's probably ok.

File: keras/backend/__init__.py [hunk 2]
```diff
@@ -83,7 +81,11 @@
...
+        sys.stderr.write('Using ' + _BACKEND + ' backend.\n')
+    except Exception:
```
[the remark this answers is not in the document]
Comment (farizrahman4u) [reply keras/backend/__init__.py]: Fixed. But you could still get an `ImportError` from "inside" the module.

**Comment by farizrahman4u** [comment]
> One possibility is to only import from the candidate module a whitelist of symbols (e.g. conv2d, etc). If all entries are there, we have a valid backend.

That would be too strict. For e.g, there could be inference-only backends that do not have `K.gradients` . It is possible to have dummy/empty methods but I think its too much. Another option is to look for a specific symbol (like `_KERAS_BACKEND`) and document this behavior.

**Comment by farizrahman4u** [comment]
@fchollet ping ping

File: keras/backend/__init__.py [hunk 3]
```diff
@@ -83,7 +82,24 @@
...
+    # Try and load external backend.
+    try:
+        backend_module = importlib.import_module(_BACKEND)
+        entries = backend_module.__dict__
+        # Check if valid backend.
+        # Module is a valid backend if it has the required entries.
+        required_entries = ['placeholder', 'variable', 'function']
+        for e in required_entries:
+            if e not in entries:
+                raise Exception('Invalid backend. Missing required entry : ' + e)
```
Comment (fchollet) [inline keras/backend/__init__.py]: This should be a `ValueError`.

Comment (farizrahman4u) [reply keras/backend/__init__.py]: done.

**Comment by farizrahman4u** [comment]
@fchollet anything else?

**Comment by fchollet** [comment]
Restarting tests.
